org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/serializer/SerializerScopeProvider.java

```diff
@@ -323,6 +324,9 @@ protected IScope getThisOrSuperScope(XAbstractFeatureCall call, JvmType thisOrSu
 			JvmDeclaredType thisType = ((JvmMember) logicalContainer).getDeclaringType();
 			if (thisType != thisOrSuper) {
 				name = SUPER;
+				if (logicalContainer instanceof JvmConstructor && thisType instanceof JvmGenericType && ((JvmGenericType) thisType).isAnonymous()) {
```

Comment:
I'd rather look into getNearestLogicalContainer: the arguments of the constructor call for an anon class constructor should not return the the JvmConstructor of the anon class as their logical container. They are resolved in the context of the enclosing type.